### PR TITLE
Issue 1457 delete sample types with deleted assets

### PR DESCRIPTION
**Pull request by kdp-cloud**
This Pull Request fixes:
- Deletes the linked sample types of the deleted assays or studies
- Takes into account the samples in the linked sample types when checking at deletion.

### CODE REVIEW

File: app/controllers/assays_controller.rb [hunk 1]
```diff
@@ -112,6 +111,13 @@ def create
     end
   end
 
+
+  def delete_linked_sample_types
+    return unless is_single_page_assay?
+
+    @assay.sample_type.delete
```
Comment (stuzart) [inline app/controllers/assays_controller.rb:118]: you should use `.destroy` to ensure any callbacks are called. `delete `will just delete the record from the database without any checks

**Review by stuzart — CHANGES_REQUESTED** [review]
I think it also needs some unit tests, to check for the model states of whether they can be deleted

File: app/controllers/assays_controller.rb (hunk 1 above)
Comment (kdp-cloud) [reply app/controllers/assays_controller.rb]: Thanks, didn't know that. See commit [eed9e1c](https://github.com/seek4science/seek/pull/1571/commits/eed9e1c917c301a49193a01bb1cea3fd3bed4ad3).

File: app/controllers/studies_controller.rb [hunk 2]
```diff
@@ -88,6 +89,17 @@ def update
     end
   end
 
+  def delete_linked_sample_types
+    return unless is_single_page_study?
+
+    @study.sample_types.each do |st|
+      raise "Sample Type '#{st.title}' contains samples. Unable to delete study" unless st.samples.empty?
+
+      st.delete
```
[the remark this answers is not in the document]
Comment (kdp-cloud) [reply app/controllers/studies_controller.rb]: See commit [eed9e1c](https://github.com/seek4science/seek/pull/1571/commits/eed9e1c917c301a49193a01bb1cea3fd3bed4ad3).

**Review by stuzart — APPROVED** [review]

**Pull request merged**